server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java

```diff
@@ -356,7 +355,7 @@ public void indexTranslogOperations(List<Translog.Operation> operations, int rec
     }
 
     private Engine.Index getIndex(final String id) {
-        final ParseContext.Document document = new ParseContext.Document();
+        final org.elasticsearch.index.mapper.Document document = new org.elasticsearch.index.mapper.Document();
```

Comment:
This is odd, in fact we have now a name clash between the lucene document class and our own. Was it better when it was part of ParseContext? Is it ok now or shall we rename the top-level class to something else that does not clash with the lucene one?

Reply:
Our version is essentially a fork, I think, and can be used as a drop-in replacement for lucene's version everywhere.  So we could keep it as `Document` and just use it in the tests where there's a name clash? Or do what we do with other forks and call it XDocument, although I think the X-prefix implies that it's a temporary fork and we plan on pushing changes upstream which I don't think is the case here.

Reply:
What would you prefer? Do you find it an improvement to have it as a top-level class with the downside of the name clash? We could also call it ESDocument if we really are concerned about the name clash

Reply:
I definitely like it as a top-level class.  ESDocument works for me.

Reply:
Cool I will wait for Julie to chime in before merging ;)

Reply:
Moving it to the top-level makes sense to me, since it's relevant and accessed even when a `ParseContext` isn't around. I'm not as sure about the name `ESDocument`, it feels very general when the class is really always meant to be used in a parsing situation, representing a parsed document. The name `ParsedDocument` is already taken though unfortunately.

Just sharing some thoughts, I don't have a strong opinion and okay with whatever you decide for a name.

Reply:
I also don't love ESDocument, maybe LuceneDocument would make more sense actually. Or we could just accept the name clash where they happen, which should not be many places.

Reply:
Thanks for bringing up ParsedDocument, I was not super familiar with this area. As far as I understand it now, ParsedDocument is more like the ES document, that can hold one or more lucene docs. So I am going with LuceneDocument if there are no objections.